### DISCUSSION TITLE
v4: How to handle 3rd party components with color props?

### DISCUSSION THREAD

**Discussion by francois-pasquier** [nativewind/nativewind#1125] [Q&A] [answered]
Is there any way to get the actual value for a variable from the loaded tailwind theme?

What is the best course of action for these kind of components?

The issue boils down to nativewind v4 doc suggesting we should be using css variables.
- Doc suggestion for theme: https://www.nativewind.dev/v4/guides/themes#dynamic-themes
- Doc suggestion for importing colors: https://www.nativewind.dev/v4/guides/themes#access-theme-values 

Doing so blocks me from exporting a colors object.

I have been scratching my head for a few hours now...

e.g: https://github.com/farhoudshapouran/react-native-ui-datepicker selectedItemColor prop

**Comment by marklawlor** [comment, marked as answer]
Use `cssInterop` to create a custom mapping. `nativeStyleToProp` can move style properties to props

```tsx
cssInterop(DateTimePicker, {
  // <...config for other props >
  selectedClassName: {
     target: false, // Don't output any other props
     nativeStyleToProp: { 
       color: "selectedItemColor" // Move any "color" values to "selectItemColor" prop
     }
  }
})
```

This will only work on native, the library author will need to adjust their component to accept a style object (not just a color) for web support.
